app/routes/dette-har-vi-gjort/$slug.tsx

```diff
@@ -0,0 +1,51 @@
+import { json, LoaderFunction, MetaFunction } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+import sanityClient from "~/sanity/sanity-client";
+
+// TODO: I propose to move this function to a shared utility/common file
+export function assertItemFound<T>(item: T | undefined): asserts item is T {
+  if (item === undefined)
+    throw new Response("Not Found", {
+      status: 404,
+    });
+}
+
+const query = (slug: string) =>
+  sanityClient.getAll("selvskryt", `slug.current == "${slug}"`);
+
+type Data = { item: Awaited<ReturnType<typeof query>>[0] };
```

Comment:
💡 Ved å dra ut data fetching til en egen funksjon kan vi utnytte type inferesen og kjøre den hele veien nedover i løypa.

Jeg foreslår det som et mulig pattern å ta i bruk. I disse banale eksemplene synes jeg det funker kjempebra, og lett og copy-paste men fortsatt ivareta full typesikkerhet 🔒

Reply:
> `Data` vs `LoaderData`

Vi har vell to alternativer.
1. `Data` som default for loaders og så bruke `ActionData` i de tilfellene vi faktisk bruker actions. (Ingen funksjonalitet jeg ser for meg vi trenger det for de eksisterende featurene 🤔 )
2. Som du foreslår gjøre navnet mer tydlig med `LoaderData` og ikke alt omfattende som `Data` er 😅 

Jeg lener litt mot enkle `Data`, men kan fint gå med på `LoaderData`, ikke så sterk mening der. 👍 

> bruke number enn 0 for å inferre typen

Ja, det er bedre 👌